src/Workspaces/Core/Portable/Execution/SerializerServiceFactory.cs

```diff
@@ -0,0 +1,20 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Composition;
+using Microsoft.CodeAnalysis.Host;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.Serialization;
+
+namespace Microsoft.CodeAnalysis.Execution
+{
+    [ExportWorkspaceServiceFactory(typeof(ISerializerService), layer: ServiceLayer.Default), Shared]
+    internal class SerializerServiceFactory : IWorkspaceServiceFactory
+    {
+        [Obsolete("This is the factory method for " + nameof(SerializerService) + ".", error: true)]
```

Comment:
Could we have our own subclass for this attribute?  it reads as if: "this is obsolete and hsould not be used.  people should transition to something else".  But, in reality, it just means:  "it's only for use by mef.  we want to keep it, but you should not reference it directly in your code."

Something like "MefConstructorAttribute : ObsoleteAttribute"?

Reply:
➡️ `ObsoleteAttribute` is sealed, but I'm considering creating a class that holds a `const string` for the message with a comment on that declaration.